### PR TITLE
Baremetal V1 API Drivers

### PR DESCRIPTION
**Pull request by elfosardo**
For #1429

This patch adds the Drivers API calls according to [Baremetal API reference](https://developer.openstack.org/api-ref/baremetal/#drivers-drivers)

List Drivers (GET)
Show driver details (GET)
Show driver properties (GET)
Show driver logical disk properties (GET)

Source code reference:
https://github.com/openstack/ironic/blob/master/ironic/api/controllers/v1/driver.py

### CODE REVIEW

File: openstack/baremetal/v1/drivers/results.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+	// A list of active hosts that support this driver
+	Hosts []string `json:"hosts"`
+
+	// Type of this driver (“classic” or “dynamic”)
+	Type string `json:"type"`
+
+	// The default boot interface used for a node with a dynamic driver,
+	// if no boot interface is specified for the node.
+	DefaultBootInterface string `json:"default_boot_interface"`
```
Comment (dtantsur) [inline openstack/baremetal/v1/drivers/results.go]: Missing BIOS

File: openstack/baremetal/v1/drivers/results.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	// The default storage interface used for a node with a dynamic driver,
+	// if no storage interface is specified for the node.
+	DefaultStorageInterface string `json:"default_storage_interface"`
+
+	// The default vendor interface used for a node with a dynamic driver,
+	// if no vendor interface is specified for the node.
+	DefaultVendorInterface string `json:"default_vendor_interface"`
+
+	// The enabled boot interfaces for this driver.
+	EnabledBootInterfaces []string `json:"enabled_boot_interfaces"`
```
Comment (dtantsur) [inline openstack/baremetal/v1/drivers/results.go]: Ditto, missing BIOS

File: openstack/baremetal/v1/drivers/results.go (hunk 1 above)
Comment (elfosardo) [reply openstack/baremetal/v1/drivers/results.go]: added

File: openstack/baremetal/v1/drivers/results.go (hunk 2 above)
Comment (elfosardo) [reply openstack/baremetal/v1/drivers/results.go]: added

**Comment by elfosardo** [comment]
@jtopjian this PR is ready for review, thank you for your comments, I pushed some changes

File: openstack/baremetal/v1/drivers/results.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	// The default storage interface used for a node with a dynamic driver,
+	// if no storage interface is specified for the node.
+	DefaultStorageInterface string `json:"default_storage_interface"`
+
+	// The default vendor interface used for a node with a dynamic driver,
+	// if no vendor interface is specified for the node.
+	DefaultVendorInterface string `json:"default_vendor_interface"`
+
+	// The enabled bios interfaces for this driver.
+	EnabledBiosInterface []string `json:"enabled_bios_interfaces"`
```
Comment (dtantsur) [inline openstack/baremetal/v1/drivers/results.go]: "Interfaces" (plural)

Comment (elfosardo) [reply openstack/baremetal/v1/drivers/results.go]: fixed

**Comment by jtopjian** [comment]
@elfosardo Let me know if/when you're ready to have this merged :)

**Comment by elfosardo** [comment]
@jtopjian this should be ready to be merged, thanks! :)

**Pull request merged**